### ISSUE TITLE
After 0 frames: loss inf @ 0.0 fps. Stats:

### ISSUE THREAD

**Issue by liuqi8827** [facebookresearch/impact-driven-exploration#6]
Thanks for your work.

I'm facing a problem when I run  `OMP_NUM_THREADS=1 python main.py --model ride --env MiniGrid-ObstructedMaze-2Dlh-v0 `
The error is :
```
Creating log directory: ../torchbeast-20201124-192823
Saving arguments to ../torchbeast-20201124-192823/meta.json
Saving messages to ../torchbeast-20201124-192823/out.log
Saving logs data to ../torchbeast-20201124-192823/logs.csv
Saving logs' fields to ../torchbeast-20201124-192823/fields.csv
[INFO:26888 ride:183 2020-11-24 19:28:24,057] Using CUDA.
Traceback (most recent call last):
  File "main.py", line 39, in <module>
  File "main.py", line 27, in main
  File "/home/lq/impact-driven-exploration/src/algos/ride.py", line 218, in train
  File "/home/lq/impact-driven-exploration/src/utils.py", line 138, in create_buffers
  File "/home/lq/anaconda3/envs/py3.6-impact-driven-exploration/lib/python3.6/site-packages/torch/tensor.py", line 330, in share_memory_
  File "/home/lq/anaconda3/envs/py3.6-impact-driven-exploration/lib/python3.6/site-packages/torch/storage.py", line 120, in share_memory_
RuntimeError: unable to open shared memory object </torch_26888_130155866> in read-write mode
```

In order to fix this problem I run `python main.py --model ride --env MiniGrid-ObstructedMaze-2Dlh-v0`
However, there is still a problem 
The INFO that I got is :
```
Creating log directory: ../torchbeast-20201124-192414
Saving arguments to ../torchbeast-20201124-192414/meta.json
Saving messages to ../torchbeast-20201124-192414/out.log
Saving logs data to ../torchbeast-20201124-192414/logs.csv
Saving logs' fields to ../torchbeast-20201124-192414/fields.csv
[INFO:26128 ride:183 2020-11-24 19:24:14,147] Using CUDA.
[INFO:26149 utils:147 2020-11-24 19:24:16,179] Actor 0 started.
[INFO:26161 utils:147 2020-11-24 19:24:16,227] Actor 1 started.
[INFO:26162 utils:147 2020-11-24 19:24:16,251] Actor 2 started.
[INFO:26174 utils:147 2020-11-24 19:24:16,283] Actor 3 started.
[INFO:26186 utils:147 2020-11-24 19:24:16,333] Actor 4 started.
[INFO:26209 utils:147 2020-11-24 19:24:16,444] Actor 5 started.
[INFO:26221 utils:147 2020-11-24 19:24:16,508] Actor 6 started.
[INFO:26233 utils:147 2020-11-24 19:24:16,564] Actor 7 started.
[INFO:26234 utils:147 2020-11-24 19:24:16,606] Actor 8 started.
[INFO:26257 utils:147 2020-11-24 19:24:16,716] Actor 9 started.
[INFO:26269 utils:147 2020-11-24 19:24:16,771] Actor 10 started.
[INFO:26281 utils:147 2020-11-24 19:24:16,890] Actor 11 started.
[INFO:26293 utils:147 2020-11-24 19:24:16,952] Actor 12 started.
[INFO:26305 utils:147 2020-11-24 19:24:17,012] Actor 13 started.
[INFO:26317 utils:147 2020-11-24 19:24:17,091] Actor 14 started.
[INFO:26329 utils:147 2020-11-24 19:24:17,172] Actor 15 started.
[INFO:26330 utils:147 2020-11-24 19:24:17,215] Actor 16 started.
[INFO:26354 utils:147 2020-11-24 19:24:17,292] Actor 17 started.
[INFO:26366 utils:147 2020-11-24 19:24:17,388] Actor 18 started.
[INFO:26378 utils:147 2020-11-24 19:24:17,468] Actor 19 started.
[INFO:26379 utils:147 2020-11-24 19:24:17,503] Actor 20 started.
[INFO:26402 utils:147 2020-11-24 19:24:17,609] Actor 21 started.
[INFO:26414 utils:147 2020-11-24 19:24:17,685] Actor 22 started.
[INFO:26426 utils:147 2020-11-24 19:24:17,788] Actor 23 started.
[INFO:26438 utils:147 2020-11-24 19:24:17,865] Actor 24 started.
[INFO:26450 utils:147 2020-11-24 19:24:17,949] Actor 25 started.
[INFO:26462 utils:147 2020-11-24 19:24:18,047] Actor 26 started.
[INFO:26475 utils:147 2020-11-24 19:24:18,104] Actor 28 started.
[INFO:26463 utils:147 2020-11-24 19:24:18,109] Actor 27 started.
[INFO:26476 utils:147 2020-11-24 19:24:18,146] Actor 29 started.
[INFO:26511 utils:147 2020-11-24 19:24:18,219] Actor 30 started.
[INFO:26523 utils:147 2020-11-24 19:24:18,340] Actor 31 started.
[INFO:26524 utils:147 2020-11-24 19:24:18,383] Actor 32 started.
[INFO:26536 utils:147 2020-11-24 19:24:18,431] Actor 33 started.
[INFO:26548 utils:147 2020-11-24 19:24:18,445] Actor 34 started.
[INFO:26549 utils:147 2020-11-24 19:24:18,486] Actor 35 started.
[INFO:26572 utils:147 2020-11-24 19:24:18,547] Actor 36 started.
[INFO:26584 utils:147 2020-11-24 19:24:18,593] Actor 37 started.
[INFO:26585 utils:147 2020-11-24 19:24:18,623] Actor 38 started.
[INFO:26586 utils:147 2020-11-24 19:24:18,657] Actor 39 started.
[INFO:26128 ride:383 2020-11-24 19:24:24,687] After 0 frames: loss inf @ 0.0 fps. Stats:
{}
[INFO:26128 ride:383 2020-11-24 19:24:29,691] After 0 frames: loss inf @ 0.0 fps. Stats:
{}
[INFO:26128 ride:383 2020-11-24 19:24:34,695] After 0 frames: loss inf @ 0.0 fps. Stats:
{}
[INFO:26128 ride:383 2020-11-24 19:24:39,699] After 0 frames: loss inf @ 0.0 fps. Stats:
{}
[INFO:26128 ride:383 2020-11-24 19:24:44,703] After 0 frames: loss inf @ 0.0 fps. Stats:
{}
[INFO:26128 ride:383 2020-11-24 19:24:49,707] After 0 frames: loss inf @ 0.0 fps. Stats:
{}
...
```

Can you give me some suggestions?
Thanks a lot!

**Comment by ptigas** [comment]
I can reproduce the same error

**Comment by ptigas** [comment]
it seems to be related to this 
```
[INFO:307 ride:383 2020-11-26 20:09:03,588] After 0 frames: loss inf @ 0.0 fps. Stats:
{}
Exception in thread batch-and-learn-3:
Traceback (most recent call last):
  File "/scratch-ssd/pankas/miniconda3/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/scratch-ssd/pankas/miniconda3/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/scratch-ssd/pankas/projects/impact-driven-exploration/src/algos/ride.py", line 320, in batch_and_learn
    inverse_dynamics_optimizer, scheduler, flags, frames=frames)
  File "/scratch-ssd/pankas/projects/impact-driven-exploration/src/algos/ride.py", line 92, in learn
    learner_outputs, unused_state = model(batch, initial_agent_state)
  File "/scratch-ssd/pankas/miniconda3/lib/python3.7/site-packages/torch/nn/modules/module.py", line 727, in _call_impl
    result = self.forward(*input, **kwargs)
  File "/scratch-ssd/pankas/projects/impact-driven-exploration/src/models.py", line 339, in forward
    F.softmax(policy_logits, dim=1), num_samples=1)
RuntimeError: probability tensor contains either `inf`, `nan` or element < 0
```

**Comment by liuqi8827** [comment]
@ptigas  Thanks.

Can anyone solve this problem?

**Comment by LotusYuki** [comment]
I fixed it by switching back to torch==1.1.0

**Comment by liuqi8827** [comment]
@LotusYuki Tanks for your help!

But if I switch back to torch==1.1.0, I have to change my CUDA version.
This will have an impact on other projects on my computer.

Do you have other suggestions?
Thanks a lot!

**Comment by BLUECARVIN** [comment]
I find a possible reason is the stored data type in buffers. The function ```create_buffers``` in ```utils.py``` stores the variable ```done``` as ```uint8``` which should be ```bool```. And in other functions, when convert it to ```float```, the code is ```(~done).float()```. In higher version pytorch, this will not correct to convert ```done``` to ```0. or 1.```. 

So my solution is modify the code in ```utils.py, line119```, ```dtype=torch.uint8``` to ```dtype=torch.bool```, and the error is gone. However, the reproducing in higher version pytorch is still need to be verified, it's need more time.

**Comment by liuqi8827** [comment]
@BLUECARVIN Thanks for pointing out the promising solution.

**Comment by maqqbu** [comment]
> @ptigas Thanks.
> 
> Can anyone solve this problem?



> @BLUECARVIN Thanks for pointing out the promising solution.

Hi, how did you solve this issue? I have the same issue when running the ride code, and I tried all the methods you mentioned here, but none of them worked.

**Comment by liuqi8827** [comment]
@maqqbu 
I fixed it by switching back to torch==1.1.0 @LotusYuki 
This project worked.

**Comment by maqqbu** [comment]
Got it, Thanks! But this method does not work for me. I finally solved it by using the method suggested here [https://github.com/facebookresearch/maskrcnn-benchmark/issues/103].
